app/src/main/java/ch/epfl/skysync/database/DelayedCallback.kt

```diff
@@ -0,0 +1,25 @@
+package ch.epfl.skysync.database
+
+/**
+ * Represents a mechanism for delaying the execution of a callback function until a certain delay
+ * has passed.
+ *
+ * This class allows for scheduling a callback function to be executed after a specified delay
+ * period. It provides functionality to decrement the delay period on each invocation of the `run()`
+ * method until the delay reaches 0, at which point the callback function is executed.
+ *
+ * @property delay The initial delay period before executing the callback, specified in units of
+ *   invocation cycles.
+ * @property callback The callback function to be executed after the delay period.
+ */
+class DelayedCallback(private var delay: Int, private val callback: () -> Unit) {
```

Comment:
I didn't really understand the use of DelayedCallback. Why did you choose this approach and generally, how do you choose the appropriate delay?

Reply:
At the moment I use the callback approach to asynchronous functions, as from what I looked the "async/await" support of Kotlin is pretty poor, the delayed callback is used when launching multiple asynchronous functions "in parallel", as each function takes a callback, but we want to execute the "final" callback only once after all functions are done. You can find an example of utilization in UserTable.deleteAvailabilities`

Reply:
got  it. Maybe update the description to something like:

_If n similar asynch. operations are executed (in a random order) but the call function should be called only once by the last of the n operations that finishes _ and change the signature to

`DelayedCallback(private var delay: Int, private val callback: () -> Unit)` => 
`ExecuteCallbackOnceByLastOperation(private var nParallelOperations: Int, private val callback: () -> Unit)`

Reply:
Renamed it to `ParallelOperationsEndCallback`